packages/drawer/src/views/DrawerItem.tsx

```diff
@@ -84,29 +88,32 @@ type Props = {
 };
 
 const LinkPressable = ({
+  route,
+  href,
   children,
   style,
   onPress,
   onLongPress,
   onPressIn,
   onPressOut,
-  to,
   accessibilityRole,
   ...rest
 }: Omit<React.ComponentProps<typeof PlatformPressable>, 'style'> & {
   style: StyleProp<ViewStyle>;
 } & {
-  to?: string;
+  route: Route<string>;
+  href?: string;
   children: React.ReactNode;
   onPress?: () => void;
 }) => {
-  if (Platform.OS === 'web' && to) {
+  if (Platform.OS === 'web') {
     // React Native Web doesn't forward `onClick` if we use `TouchableWithoutFeedback`.
     // We need to use `onClick` to be able to prevent default browser handling of links.
     return (
       <Link
         {...rest}
-        to={to}
+        href={href}
+        action={CommonActions.navigate(route.name, route.params)}
```

Comment:
If require `route` here we could build `buildHref(route.name, route.params)` right here as well in place rather than prop drilling it a few levels